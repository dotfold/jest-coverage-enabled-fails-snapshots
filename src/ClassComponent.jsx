
import React from 'react'
import FunctionalComponent from './FunctionalComponent'
import AnotherClassComponent from './AnotherClassComponent'

class ClassComponent extends React.PureComponent {
  render () {
    return (
      <div>
        This is a Class Component that contains a FunctionalComponent - this one fails the snapshot after enabling coverage
        <FunctionalComponent />
        And here is AnotherClassComponent - this one is ok in the snapshot
        <AnotherClassComponent />
      </div>
    )
  }
}

export default ClassComponent
