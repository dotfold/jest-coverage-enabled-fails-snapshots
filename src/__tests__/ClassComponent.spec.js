
import React from 'react'
import renderer from 'react-test-renderer'

import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import ClassComponent from '../ClassComponent'

describe('<ClassComponent />', () => {
  it('should render correctly with react-test-renderer', () => {
    const wrapper = renderer.create(
      <ClassComponent />
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly with enzyme', () => {
    const wrapper = shallow(
      <ClassComponent />
    )
    console.log(wrapper.debug())
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
