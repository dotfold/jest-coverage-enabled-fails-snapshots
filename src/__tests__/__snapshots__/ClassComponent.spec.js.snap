exports[`<ClassComponent /> should render correctly with enzyme 1`] = `
<div>
  This is a Class Component that contains a FunctionalComponent - this one fails the snapshot after enabling coverage
  <FunctionalComponent />
  And here is AnotherClassComponent - this one is ok in the snapshot
  <AnotherClassComponent />
</div>
`;

exports[`<ClassComponent /> should render correctly with react-test-renderer 1`] = `
<div>
  This is a Class Component that contains a FunctionalComponent - this one fails the snapshot after enabling coverage
  <div>
    This is a functional component
  </div>
  And here is AnotherClassComponent - this one is ok in the snapshot
  <div>
    This is a Class Component that contains a FunctionalComponent
    <div>
      This is a functional component
    </div>
  </div>
</div>
`;
